### PR TITLE
sectransp: support CURLINFO_CERTINFO

### LINKED ISSUE #4130: curl built with DarwinSSL doesn't collect server cert
### I did this
./buildconfig
./configure --withdarwin-ssl
make
cd docs/examples
make && certinfo

### I expected the following
1 certs!
...some cert data here...

### curl/libcurl version

built from the latest master  (aa73eb47bc8583070734696b25b34ad54c2c1f5e).

### operating system

macOS 10.14.5

### PR DESCRIPTION
**Pull request by sergio-nsk**
I need it for verifying certificate chains with root certificates missing in system storages.

Fixes #4130

Also fixed compiler warning: unused buffer, when the option CURL_DISABLE_VERBOSE_STRINGS is ON.

### CODE REVIEW

File: docs/libcurl/opts/CURLINFO_CERTINFO.3 [hunk 1]
```diff
@@ -71,8 +71,9 @@ if(curl) {
 }
 .fi
 .SH AVAILABILITY
-This option is only working in libcurl built with OpenSSL, NSS, Schannel or
-GSKit support. Schannel support added in 7.50.0
+This option is only working in libcurl built with OpenSSL, NSS, Schannel, GSKit
+or Secure Transport support. Schannel support added in 7.50.0. Secure Transport
+support added in 7.78.0.
```
Comment (danielgustafsson) [inline docs/libcurl/opts/CURLINFO_CERTINFO.3:76]: This won't make it into 7.78.0, so it should sat 7.79.0. That being said, I think we need a more structured way to document the version availability than free text like this - that's not the responsibility of this patch but it's worth to start thinking about.

File: lib/connect.c [hunk 2]
```diff
@@ -1092,9 +1092,11 @@ static void nosigpipe(struct Curl_easy *data,
   int onoff = 1;
   if(setsockopt(sockfd, SOL_SOCKET, SO_NOSIGPIPE, (void *)&onoff,
                 sizeof(onoff)) < 0) {
+#if !defined(CURL_DISABLE_VERBOSE_STRINGS)
     char buffer[STRERROR_LEN];
     infof(data, "Could not set SO_NOSIGPIPE: %s",
           Curl_strerror(SOCKERRNO, buffer, sizeof(buffer)));
+#endif
```
Comment (danielgustafsson) [inline lib/connect.c:1099]: This seems unrelated?

Comment (sergio-nsk) [reply lib/connect.c:1099]: Yes, this is unrelated. It fixes the warning unused local variable `buffer` when I compile with cmake option `-DCURL_DISABLE_VERBOSE_STRINGS=ON`. I can remove this change it from this PR.

**Comment by danielgustafsson** [comment]
> Yes, this is unrelated. It fixes the warning unused local variable buffer when I compile with cmake option -DCURL_DISABLE_VERBOSE_STRINGS=ON. I can remove this change it from this PR.

Please do, and file it as its own PR. That smells like a bug that we should fix in 7.78 already.

**Comment by sergio-nsk** [comment]
> Please do, and file it as its own PR. That smells like a bug that we should fix in 7.78 already.

See #7377.

Updated the supported version.

**Comment by bagder** [comment]
Can you please rebase this (and fix the merge conflicts) and force-push?

**Comment by bagder** [comment]
The sectransp CI build failed:
~~~
vtls/sectransp.c:2975:66: error: expected ')'
        result = collect_server_cert_single(data, server_cert, i);
                                                                 ^
vtls/sectransp.c:2974:64: note: to match this '('
        server_cert = (SecCertificateRef)CFArrayGetValueAtIndex(server_certs,
                                                               ^
vtls/sectransp.c:2957:47: error: implicit conversion loses integer precision: 'CFIndex' (aka 'long') to 'int' [-Werror,-Wshorten-64-to-32]
        result = Curl_ssl_init_certinfo(data, count);
                 ~~~~~~~~~~~~~~~~~~~~~~       ^~~~~
vtls/sectransp.c:2972:47: error: implicit conversion loses integer precision: 'CFIndex' (aka 'long') to 'int' [-Werror,-Wshorten-64-to-32]
        result = Curl_ssl_init_certinfo(data, count);
                 ~~~~~~~~~~~~~~~~~~~~~~       ^~~~~

**Comment by sergio-nsk** [comment]
It seems I've broken the code while fixing the merge conflicts.

**Comment by bagder** [comment]
Thanks!
